### PR TITLE
refactor: Port renderer-internal to TypeScript

### PR DESCRIPTION
**Pull request by felixrieseberg**
Another chunk of TypeScript conversion, this time around for `renderer-internal`.

Notes: no-notes

### CODE REVIEW

File: lib/renderer/ipc-renderer-internal.ts [hunk 1]
```diff
@@ -1,10 +1,8 @@
-'use strict'
-
 const binding = process.atomBinding('ipc')
 const v8Util = process.atomBinding('v8_util')
 
 // Created by init.js.
-const ipcRenderer = v8Util.getHiddenValue(global, 'ipc-internal')
+export const ipcRenderer: Electron.IpcRenderer = v8Util.getHiddenValue(global, 'ipc-internal')
```
Comment (MarshallOfSound) [inline lib/renderer/ipc-renderer-internal.ts]: Can we copy the main process here and `export const ipcRendererInternal: ...` to enforce a naming convention in our code 👍 (So we don't get confused between `ipcRenderer` and `ipcRendererInternal` 👍 )

Comment (felixrieseberg) [reply lib/renderer/ipc-renderer-internal.ts]: Good callout! I've updated the whole PR.

**Comment by MarshallOfSound** [comment]
Few TSC errors now 👍

**Comment by felixrieseberg** [comment]
`**build-mac **` passed, it just failed in `mksnapshot` for `mas` again, which is unrelated 😬

**Pull request merged**